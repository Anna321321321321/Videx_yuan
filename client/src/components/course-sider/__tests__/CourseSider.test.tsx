import { shallow } from 'enzyme';
import React from 'react';
import CourseSider from '..';

const courses = [
  {
    name: 'test1',
    id: '1'
  },
  {
    name: null,
    id: '2'
  }
];

describe('CourseSider Unit Tests', () => {
  it('Render view of student', () => {
    const wrapper = shallow(
      <CourseSider courses={courses} adminAccess={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Render view of teacher', () => {
    const wrapper = shallow(
      <CourseSider courses={courses} adminAccess={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
