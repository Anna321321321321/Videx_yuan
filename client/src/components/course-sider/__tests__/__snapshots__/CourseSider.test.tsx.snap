// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseSider Unit Tests Render view of student 1`] = `
<Sider
  breakpoint="md"
  collapsedWidth={0}
  collapsible={false}
  defaultCollapsed={false}
  prefixCls="ant-layout-sider"
  reverseArrow={false}
  style={
    Object {
      "background": "#fff",
    }
  }
  theme="dark"
  width={300}
>
  <Menu
    className=""
    defaultOpenKeys={
      Array [
        "manage",
        "courses",
      ]
    }
    focusable={false}
    inlineIndent={12}
    mode="inline"
    prefixCls="ant-menu"
    style={
      Object {
        "borderRight": 0,
        "height": "100%",
      }
    }
    theme="light"
  >
    <SubMenu
      key="courses"
      title={
        <span>
          <Icon
            type="book"
          />
          Courses
        </span>
      }
    >
      <MenuItem
        key="0"
      >
        <Link
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
            }
          }
          to="/course/1"
        >
          test1
        </Link>
      </MenuItem>
    </SubMenu>
  </Menu>
</Sider>
`;

exports[`CourseSider Unit Tests Render view of teacher 1`] = `
<Sider
  breakpoint="md"
  collapsedWidth={0}
  collapsible={false}
  defaultCollapsed={false}
  prefixCls="ant-layout-sider"
  reverseArrow={false}
  style={
    Object {
      "background": "#fff",
    }
  }
  theme="dark"
  width={300}
>
  <Menu
    className=""
    defaultOpenKeys={
      Array [
        "manage",
        "courses",
      ]
    }
    focusable={false}
    inlineIndent={12}
    mode="inline"
    prefixCls="ant-menu"
    style={
      Object {
        "borderRight": 0,
        "height": "100%",
      }
    }
    theme="light"
  >
    <SubMenu
      key="manage"
      title={
        <span>
          <Icon
            type="solution"
          />
          Manage
        </span>
      }
    >
      <MenuItem
        key="create"
      >
        <Link
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
          to="course/create"
        >
          <Icon
            type="file-add"
          />
          New Course
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      key="courses"
      title={
        <span>
          <Icon
            type="book"
          />
          Courses
        </span>
      }
    >
      <MenuItem
        key="0"
      >
        <Link
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
            }
          }
          to="/course/1"
        >
          test1
        </Link>
      </MenuItem>
    </SubMenu>
  </Menu>
</Sider>
`;
